src/lib/viewers/text/TextBaseViewer.js

```diff
@@ -143,6 +147,30 @@ class TextBaseViewer extends BaseViewer {
         this.controls.add(__('exit_fullscreen'), this.toggleFullscreen, 'bp-exit-fullscreen-icon', ICON_FULLSCREEN_OUT);
     }
 
+    loadUIReact() {
+        this.controls = new ControlsRoot({ containerEl: this.containerEl });
+        this.renderUI();
+    }
+
+    renderUI() {
+        if (this.zoomControls) {
```

Comment:
Do we need to include this in this code path? Seems like once we enable `useReactControls` zoomControls wouldn't be needed?

Reply:
It's used in both code paths when the viewer state changes to avoid having the conditions spread out everywhere.